rpc/estimate_fee.go

```diff
@@ -54,18 +89,22 @@ func (h *Handler) EstimateFee(broadcastedTxns []BroadcastedTransaction,
 	}), httpHeader, nil
 }
 
-func (h *Handler) EstimateMessageFee(msg MsgFromL1, id BlockID) (*FeeEstimate, http.Header, *jsonrpc.Error) { //nolint:gocritic
-	return h.estimateMessageFee(msg, id, h.EstimateFee)
+//nolint:gocritic
+func (h *Handler) EstimateMessageFeeV0_7(msg MsgFromL1, id BlockID) (*FeeEstimateV0_7, http.Header, *jsonrpc.Error) {
+	return estimateMessageFee(msg, id, h.EstimateFeeV0_7)
+}
+
+//nolint:gocritic
+func (h *Handler) EstimateMessageFee(msg MsgFromL1, id BlockID) (*FeeEstimate, http.Header, *jsonrpc.Error) {
+	return estimateMessageFee(msg, id, h.EstimateFee)
 }
 
-type estimateFeeHandler func(broadcastedTxns []BroadcastedTransaction,
+type estimateFeeHandler[T any] func(broadcastedTxns []BroadcastedTransaction,
 	simulationFlags []SimulationFlag, id BlockID,
-) ([]FeeEstimate, http.Header, *jsonrpc.Error)
+) ([]T, http.Header, *jsonrpc.Error)
```

Comment:
I don't think the usage of generics provides any significant benefits here; rather, it adds more code complexity. I'd suggest the following approach instead:

1. Create an internal function that both `EstimateMessageFee` methods use in common
2. In `EstimateMessageFeeV0_7`, convert the output to the `V0_7` format

A simple code example would be:
```
func (h *Handler) EstimateMessageFeeV0_7(msg MsgFromL1, id BlockID) (*FeeEstimateV0_7, http.Header, *jsonrpc.Error) {
    estimate, header, err := h.estimateMessageFee(msg, id)
    if err != nil {
        return nil, header, err
    }
    result := FeeEstimateToV0_7(estimate)
    return &result, header, nil
}

func (h *Handler) EstimateMessageFee(msg MsgFromL1, id BlockID) (*FeeEstimate, http.Header, *jsonrpc.Error) {
    estimate, header, err := h.estimateMessageFee(msg, id)
    if err != nil {
        return nil, header, err
    }
    return &estimate, header, nil
}

func (h *Handler) estimateMessageFee(msg MsgFromL1, id BlockID) (FeeEstimate, http.Header, *jsonrpc.Error) {
   ...whatever that is common
}
```